app/services/facebook_authenticator.rb

```diff
@@ -23,27 +26,30 @@ def authenticate!
 
   private
 
-    def login
-      @login ||= Login.where(identification: facebook_user[:email]).first
+    def find_login(facebook_user)
+      Login.where(identification: facebook_user[:email]).first
     end
 
-    def connect_login_to_fb_account
+    def connect_login_to_fb_account(login, facebook_user)
       login.update_attributes!(facebook_uid: facebook_user[:id])
     end
 
-    def create_login_from_fb_account
+    def create_login_from_fb_account(facebook_user)
       login_attributes = {
         identification: facebook_user[:email],
         facebook_uid:   facebook_user[:id]
       }
 
-      @login = Login.create!(login_attributes)
+      Login.create!(login_attributes)
+    end
+
+    def access_token
+      response = facebook_request(fb_token_url(@auth_code)).body
+      response.match(/access_token=([^&]+)/)[1]
```

Comment:
should be fine to use `access_token` then

Marco Otte-Witte
marco.otte-witte@simplabs.com
http://simplabs.com

> Am 08.09.2015 um 14:23 schrieb Oliver Azevedo Barnes <notifications@github.com>:
> 
> In app/services/facebook_authenticator.rb <https://github.com/simplabs/rails_api_auth/pull/18#discussion_r38918762>:
> 
> >        login_attributes = {
> >          identification: facebook_user[:email],
> >          facebook_uid:   facebook_user[:id]
> >        }
> >  
> > -      @login = Login.create!(login_attributes)
> > +      Login.create!(login_attributes)
> > +    end
> > +
> > +    def access_token
> > +      response = facebook_request(fb_token_url(@auth_code)).body
> > +      response.match(/access_token=([^&]+)/)[1]
> rubocop doesn't like get_access_token without an argument. Sees it as a reader method using a get_ prefix, which is bad form usually. maybe I can keep the name and disable that cop
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/simplabs/rails_api_auth/pull/18/files#r38918762>.
>